libs/ui/src/lib/components/login-modal.tsx

```diff
@@ -0,0 +1,86 @@
+'use client';
+
+import Image from 'next/image';
+import { Modal } from 'antd';
+import { Dispatch, SetStateAction } from 'react';
+import { GithubButton } from './button/github';
+import { GoogleButton } from './button/google';
+
+interface LoginModalProps {
+  isOpen: boolean;
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
+}
+
+interface ImageProps {
+  source: string;
+  altText: string;
+}
+
+export default function LoginModal({ isOpen, setIsOpen }: LoginModalProps) {
+  function closeModal() {
+    setIsOpen(false);
+    document.body.style.overflow = 'auto';
+  }
+
+  function generateRandomImage() {
+    const images = [
+      {
+        source: '/login.svg',
+        altText: 'Garota acessando sua conta em um celular.',
+      },
+      {
+        source: '/login2.svg',
+        altText: 'Mulher acessando sua conta em um celular.',
+      },
+      {
+        source: '/login3.svg',
+        altText: 'Garoto colocando uma chave dentro de um cadeado, simbolizando a senha.',
+      },
+      {
+        source: '/login4.svg',
+        altText: 'Garota acessando sua conta em um celular.',
+      },
+      {
+        source: '/login5.svg',
+        altText: 'Garoto e garota acessando suas contas em um celular.',
+      },
+    ];
+
+    const previousImages: ImageProps[] = [];
+
+    const availableImages = images.filter((image) => !previousImages.includes(image));
+
+    if (availableImages.length === 0) {
+      return images[Math.floor(Math.random() * 5)];
+    }
+
+    const randomIndex = Math.floor(Math.random() * availableImages.length);
+    const randomImage = availableImages[randomIndex];
+
+    previousImages.push(randomImage);
+
+    return randomImage;
+  }
```

Comment:
I think this function can be a hook `useRandomImage`. 

### create the file in `ui/src/lib/hooks/random-image.ts`
example file: 
```ts
import { Image } from '../interfaces';

export const useRandomImage = (images: Image[]) => {
  function nextImage() {
    const previousImages: Image[] = [];

    const availableImages = images.filter((image) => !previousImages.includes(image));

    if (availableImages.length === 0) {
      return images[Math.floor(Math.random() * 5)];
    }

    const randomIndex = Math.floor(Math.random() * availableImages.length);
    const randomImage = availableImages[randomIndex];

    previousImages.push(randomImage);

    return randomImage;
  }

  return [nextImage];
};
```
### using hook: 
```ts
  const [nextImage] = useRandomImage(images);
  const randomImage = nextImage();
```

And image array can be substituted  out to of the component. 

Like this: 
```ts
const images = [
  {
    source: '/login.svg',
    altText: 'Garota acessando sua conta em um celular.',
  },
  {
    source: '/login2.svg',
    altText: 'Mulher acessando sua conta em um celular.',
  },
  {
    source: '/login3.svg',
    altText: 'Garoto colocando uma chave dentro de um cadeado, simbolizando a senha.',
  },
  {
    source: '/login4.svg',
    altText: 'Garota acessando sua conta em um celular.',
  },
  {
    source: '/login5.svg',
    altText: 'Garoto e garota acessando suas contas em um celular.',
  },
];

export default function LoginModal({ isOpen, setIsOpen }: LoginModalProps) {
```